src/sass/colors/_colors-helpers.scss

```diff
@@ -32,9 +32,32 @@ Styleguide Colors.Helpers
   .background-#{$color} {
     background-color: $colorValue;
   }
-  .btn-#{$color} {
-    color: $white;
-    background-color: $colorValue;
+  .btn-#{$color} { // Respect button contrast
+    @if (str-index($color, 'white') or str-index($color, 'light') or str-index($color, 'lighter') or str-index($color, 'smoke') ) {
+      color: $grey-darker !important;
```

Comment:
In this case, `!important` is intended so button color contrast by definition is always respected 
- Darker buttons will always have white icons and font
- Lighter buttons will always have black(grey-darker) fonts
So this means, If I add a `btn btn-warning` inside any structure, (navbar, settings menu, floating modal, pricing box) it's contrast will be respected, so no we never have to go custom buttons.
This is to ensure e.g.`btn btn-primary` doesn't get black font/icons on any scenario and avoid stuff like:
`body.settings { 
   &.btn.btn-primary {
      color: white;
      &.icon{
          fill: white;
      }
   }
}`